src/codegen/riscv64/macro-assembler-riscv64.cc

```diff
@@ -3645,17 +3645,20 @@ void TurboAssembler::MulOverflow(Register dst, Register left,
          overflow != scratch);
   DCHECK(overflow != left && overflow != right_reg);
 
+
+  addiw(left, left, 0);
+  addiw(right_reg, right_reg, 0);
+
   if (dst == left || dst == right_reg) {
     Mul(scratch, left, right_reg);
-    Mulh(overflow, left, right_reg);
+    Dmul(overflow, left, right_reg);
```

Comment:
Let's use the `Assembler` interfaces directly here instead (`mul`, `mulw`).

Reply:
I agree: it looks like these high level mul routines came from MIPS and should probably be cleaned up long term, so it's better to just use the actual instructions.

Reply:
Thank you, I will fix it.